src/node/txreconciliation.cpp

```diff
@@ -81,31 +82,28 @@ class TxReconciliationTracker::Impl
     {
         AssertLockNotHeld(m_txreconciliation_mutex);
         LOCK(m_txreconciliation_mutex);
-        // We do not support txreconciliation salt/version updates.
-        assert(m_states.find(peer_id) == m_states.end());
 
         LogPrintLevel(BCLog::TXRECONCILIATION, BCLog::Level::Debug, "Pre-register peer=%d\n", peer_id);
         const uint64_t local_salt{GetRand(UINT64_MAX)};
 
         // We do this exactly once per peer (which are unique by NodeId, see GetNewNodeId) so it's
         // safe to assume we don't have this record yet.
-        Assert(m_states.emplace(peer_id, local_salt).second);
+        Assume(m_states.emplace(peer_id, local_salt).second);
         return local_salt;
     }
 
-    ReconciliationRegisterResult RegisterPeer(NodeId peer_id, bool is_peer_inbound, bool is_peer_recon_initiator,
-                                     bool is_peer_recon_responder, uint32_t peer_recon_version,
+    ReconciliationRegisterResult RegisterPeer(NodeId peer_id, bool is_peer_inbound, uint32_t peer_recon_version,
                                      uint64_t remote_salt) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
     {
         AssertLockNotHeld(m_txreconciliation_mutex);
         LOCK(m_txreconciliation_mutex);
         auto recon_state = m_states.find(peer_id);
 
         // A peer should be in the pre-registered state to proceed here.
-        if (recon_state == m_states.end()) return NOT_FOUND;
+        if (recon_state == m_states.end()) return ReconciliationRegisterResult::NOT_FOUND;
         uint64_t* local_salt = std::get_if<uint64_t>(&recon_state->second);
         // A peer is already registered. This should be checked by the caller.
-        Assume(local_salt);
+        assert(local_salt);
```

Comment:
There is some inconsistency between handling these edge cases. This `assert()` also looks a bit dangerous to me wrt to future changes - it seems that if the peer sends us two times `SENDTXRCNCL` that could trigger a remote crash by tripping this `assert()`. The only thing that is stopping this now is the `IsPeerRegistered()` in the caller, but it is not obvious there that if it is removed, grave things will ensue. I can imagine it getting wiped away by a future "refactor".

If the caller has checked `IsPeerRegistered()` before this, then we will never return `NOT_FOUND` from here either. So that may as well be an assert too :fire: 

What about not having such an assert at all and returning `ReconciliationRegisterResult::NOT_FOUND` if the peer is not pre-registered and `ReconciliationRegisterResult::ALREADY_REGISTERED` if the peer is already registered? Then the caller need not to check `IsPeerRegistered()` in before calling this method.